### DISCUSSION TITLE
gaussian blur in your program

### DISCUSSION THREAD

**Discussion by andy500** [nagadomi/nunif#254] [Ideas]
can you put in your program  a gaussian blur as it would reduce the background artifacts errors

**Comment by nagadomi** [comment]
I am not sure what "background artifacts errors" are.
If you mean artifacts at the boundary between background and foreground, I have already did that with `Edge Fix` option.
https://github.com/nagadomi/nunif/blob/master/iw3/README.md#what-is---edge-dilation-option

**Comment by andy500** [comment]
no it is a person moving in the foreground and the background stick to to him as he moves say he moves pass window frame it bene the frame as he pass it all any object in the background

**Reply by andy500** [reply]
that is why you use gaussian blur in depth maps it reduces the artifacts errors in the background

**Reply by nagadomi** [reply]
It is a problem of model, it is difficult to improve the average quality of any scene by simple post processing.

**Reply by donrikk** [reply]
i have produced artifact free results using your program by blurring mid and background in maps while keeping foreground whites sharp as they are supossed to then i do a edge dilation after and that seems to fix that issue so andy is definitely not lying to you on this one gaussian blurring or 2d gaussian splatting for the mid and background on map export will in fact help this issue, it may not be 100 percent perfect all the time but majority of your film will be free of those kind of artifacts. it also helps to being everything into the grey level pertaining to background and midground making blacks come into the greys into the map instead of having background as black, if you set the depth strength in your program to 2.5 to 3.0 depth strength you come out with minimal to no artifacts on result and is very very similar to bluray release conversions.

**Reply by nagadomi** [reply]
Does the artifact refer to the artifact at the boundary between the background and foreground in stereo generation (row_flow_v3)? Not referring to the results of the depth estimation model.

**Reply by donrikk** [reply]
yes, that edge warping that occurs on the final output in some places when disparity is more drastic it causes things in the background and mid ground to kind of bend into the foreground objects edge usually more noticeable if the background has a vertical or horizontal line of some sort, more abstract backgrounds like sky and so on dont cause a big concern but vertical nd horizontal lines as u probably already know tend to do that bending warped effect which already isnt to bad as is. but when you do the blurring of midground and foreground while keeping the whites sharper in the map and then do the dilation to the maps after that it tends to be much better. i created a depth blender that blends the v1 maps into the v2 maps that does this blurring to the map outputs here is the python script for it in txt document if you would like to take a look its a very fast process and can probably be implemented with little to know performance or speed hit to the end user. 
[depth blender.txt](https://github.com/user-attachments/files/17792942/depth.blender.txt)

**Reply by donrikk** [reply]
blending the maps with this blender fixes missing whites in the v2 maps during blurred objects in scene or what ever other case it may do it, while retaining the detail of v2 without any increased banding on resize either. this does mean i haveto produce both v1 and v2 maps as a sequence to blend them but overall the wait is about the same as it normlly would be even with my editing

**Reply by nagadomi** [reply]
Then I think what is needed is to improve Edge Fix/row_flow_v3, I think it is no need to blur the entire background (lower depth value area).
Ensemble method is another topic.

**Reply by donrikk** [reply]
Yes I think if u added specific Gaussian blurring perameters into the edge fix process itself it can do it while it’s doing the dilation which is what most of us do when editing our maps anyways, having it as a baked in option on edge fix in app would save a lot of time and effort though for sure the blur would slightly blur the whites and slightly more aggressively to the greys and black. In my blender script I sent here u can take a look at the blurring perameters in it and see if it’s something that could be thrown into the edge fixing code.

**Reply by donrikk** [reply]
But if u can fix row flow then it may be way more beneficial than blurring. If that’s possible ide def stick with that

**Reply by nagadomi** [reply]
`row_flow_v3` was trained before DepthAnything V2 was released so V2 depth is not used for training.
It may improve just by re-training.
However, it uses the result of `forward_fill` as the ground truth data, so if `forward_fill` produces similar artifacts, it cannot be fixed.

**Reply by andy500** [reply]
donrikk where does your python script depth blender go in the program

**Reply by donrikk** [reply]
> donrikk where does your python script depth blender go in the program

https://youtu.be/ttUZHGXVxts?si=Lbpm188MAGXMVLGU

That YouTube link will bring to my tutorial on how to do my map edits it’ll show the whole process including where everything goes and what’s supposed to be there in the folder set ups for the scripts.

**Reply by donrikk** [reply]
For the blender u want to put the code in any folder then create 2 folders one called v1 and the other called v2 then when u produce ur maps put ur v1 maps in that v1 folder and your v2 maps in the v2 folder, u then run the s riot it’ll come up with a gui u then enter ur new depth maps size i usually go with the video source res so i resize from 910x518 to 1920x1080 it uses lances and blurring so there is no added banding on output. The output blended maps will replace the corresponding map on the v2 folder so it’ll overwrite the v2 maps with the new blended ones.

**Reply by andy500** [reply]
l just try Owl3D and it as a option so you can convert the left eye only and it works as l have no artifacts in the background with the movie  l have had errors in the background Forbidden.Planet so if you could create a way of convert the left eye all the right eye only
